Source/Core/UICommon/DiscordPresence.cpp

```diff
@@ -15,20 +19,69 @@
 
 namespace Discord
 {
+#ifdef USE_DISCORD_PRESENCE
+static JoinFunction join_function = nullptr;
+static const char* username = "";
+
+static void HandleDiscordReady(const DiscordUser* user)
+{
+  username = user->username;
+}
+
+static void HandleDiscordJoin(const char* join_secret)
+{
+  if (join_function != nullptr)
+  {
+    if (Config::Get(Config::NETPLAY_NICKNAME) == Config::NETPLAY_NICKNAME.default_value)
+      Config::SetBaseOrCurrent(Config::NETPLAY_NICKNAME, username);
+
+    std::string secret(join_secret);
+    std::string host = secret.substr(0, secret.find_last_of(":"));
+    Config::SetBaseOrCurrent(Config::NETPLAY_HOST_CODE, host);
```

Comment:
You need to check whether this value is an IP address or host code, and set either NETPLAY_ADDRESS or NETPLAY_HOST_CODE respectively, as well as change NETPLAY_TRAVERSAL_CHOICE to "direct" or "traversal" respectively.